// admin-bro/lib/frontend/components/app/base-action-component.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactI18next = require("react-i18next");

var _errorBoundary = _interopRequireDefault(require("./error-boundary"));

var actions = _interopRequireWildcard(require("../actions"));

var _constants = require("../../../constants");

var _designSystem = require("../design-system");

var _hooks = require("../../hooks");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Component which renders all the default and custom actions for both the Resource and the Record.
 *
 * It passes all props down to the actual Action component.
 *
 * Example of creating your own actions:
 * ```
 * // AdminBro options
 * const AdminBroOptions = {
 *   resources: [
 *      resource,
 *      options: {
 *        actions: {
 *           myNewAction: {
 *             label: 'amazing action',
 *             icon: 'Add',
 *             inVisible: (resource, record) => record.param('email') !== '',
 *             actionType: 'record',
 *             component: AdminBro.bundle('./my-new-action'),
 *             handler: (request, response, data) => {
 *               return {
 *                  ...
 *               }
 *             }
 *           }
 *        }
 *      }
 *   ]
 * }
 * ```
 *
 * ```
 * // ./my-new-action.jsx
 * import { Box } from 'admin-bro'
 *
 * const MyNewAction = (props) => {
 *   const { resource, action, record } = props
 *   // do something with the props and render action
 *   return (
 *     <Box>Some Action Content</Box>
 *   )
 * }
 * ```
 *
 * @component
 * @name BaseActionComponent
 * @subcategory Application
 */
const BaseActionComponent = props => {
  const {
    resource,
    action,
    record,
    records,
    setTag
  } = props;
  const documentationLink = [_constants.DOCS, 'BaseAction.html'].join('/');
  const {
    translateMessage
  } = (0, _hooks.useTranslation)();
  let Action = actions[action.name];

  if (action.component) {
    Action = AdminBro.UserComponents[action.component];
  }

  if (Action) {
    return _react.default.createElement(_errorBoundary.default, null, _react.default.createElement(Action, {
      action: action,
      resource: resource,
      record: record,
      records: records,
      setTag: setTag
    }));
  }

  return Action || _react.default.createElement(_designSystem.MessageBox, {
    variant: "danger"
  }, translateMessage('noActionComponent'), _react.default.createElement(_reactI18next.Trans, {
    key: "messages.buttons.seeTheDocumentation"
  }, "See:", _react.default.createElement(_designSystem.Link, {
    ml: "default",
    href: documentationLink
  }, "the documentation")));
};

var _default = BaseActionComponent;
exports.default = _default;
