// admin-bro/lib/frontend/components/design-system/atoms/table/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Table", {
  enumerable: true,
  get: function () {
    return _table.default;
  }
});
Object.defineProperty(exports, "TableHead", {
  enumerable: true,
  get: function () {
    return _tableHead.default;
  }
});
Object.defineProperty(exports, "TableBody", {
  enumerable: true,
  get: function () {
    return _tableBody.default;
  }
});
Object.defineProperty(exports, "TableCell", {
  enumerable: true,
  get: function () {
    return _tableCell.default;
  }
});
Object.defineProperty(exports, "TableRow", {
  enumerable: true,
  get: function () {
    return _tableRow.default;
  }
});
Object.defineProperty(exports, "TableCaption", {
  enumerable: true,
  get: function () {
    return _tableCaption.default;
  }
});

var _table = _interopRequireDefault(require("./table"));

var _tableHead = _interopRequireDefault(require("./table-head"));

var _tableBody = _interopRequireDefault(require("./table-body"));

var _tableCell = _interopRequireDefault(require("./table-cell"));

var _tableRow = _interopRequireDefault(require("./table-row"));

var _tableCaption = _interopRequireDefault(require("./table-caption"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
